### PR TITLE
Change User Manual Description on ARMA with internalParallel option

### LINKED ISSUE #597: ARMA reseeding not as expected
--------
Issue Description
--------
##### What did you expect to see happen?
Seeding of ARMA ROM should be controlled independent of other seeds (Monte Carlo samplers, etc).

##### What did you see instead?
RNG is global in RAVEN, so seeding is for the entire workflow, not individual parts.

##### Do you have a suggested fix for the development team?
Move RNG from global to local implementation, so entities needing random numbers can be distinct but reproducible.

The ARMA reseed test needs this improvement to be consistent.

----------------
For Change Control Board: Issue Review
----------------
This review should occur before any development is performed as a response to this issue.
- [x] 1. Is it tagged with a type: defect or improvement?
- [x] 2. Is it tagged with a priority: critical, normal or minor?
- [x] 3. If it will impact requirements or requirements tests, is it tagged with requirements?
- [x] 4. If it is a defect, can it cause wrong results for users? If so an email needs to be sent to the users.
- [x] 5. Is a rationale provided? (Such as explaining why the improvement is needed or why current code is wrong.)

-------
For Change Control Board: Issue Closure
-------
This review should occur when the issue is imminently going to be closed.
- [x] 1. If the issue is a defect, is the defect fixed?
- [x] 2. If the issue is a defect, is the defect tested for in the regression test system?  (If not explain why not.)
- [x] 3. If the issue can impact users, has an email to the users group been written (the email should specify if the defect impacts stable or master)?
- [x] 4. If the issue is a defect, does it impact the latest stable branch? If yes, is there any issue tagged with stable (create if needed)?
- [x] 5. If the issue is being closed without a merge request, has an explanation of why it is being closed been provided?

### PR DESCRIPTION
**Pull request by JiaZhou-PU**
--------
Pull Request Description
--------
##### What issue does this change request address? (Use "#" before the issue to link it, i.e., #42.)
Signal generated by ARMA with ressedcopies=False and with internalParallel=true will be identical regardless of how many samples have been set up. Partly closes #597 

##### What are the significant changes in functionality due to this change request?
Change reseed raven framework test, so it can test reseedcopies options. Also documented in user manual.

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code.
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [x] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [x] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [x] 8. If an analytic test is changed/added is the the analytic documentation updated/added?
- [x] 9. If any test used as a basis for documentation examples (currently found in `raven/tests/framework/user_guide` and `raven/docs/workshop`) have been changed, the associated documentation must be reviewed and assured the text matches the example.

### CODE REVIEW

**Comment by JiaZhou-PU** [comment]
@PaulTalbot-INL Ready for review. I am still working on my English...

**Comment by PaulTalbot-INL** [comment]
> @PaulTalbot-INL Ready for review. I am still working on my English...

Aren't we all!  I'll start looking at it.

File: tests/framework/ROM/TimeSeries/ARMA/arma_reseed_precursor.xml [hunk 1]
```diff
@@ -7,20 +7,21 @@
...
-      this run only creates the arma and pickles it.
+      this run creates the arma and pickles it, also generates gold files for the reseed test.
     </description>
     <revisions>
       <revision author="talbpaul" date="2017-07-12">created</revision>
       <revision author="alfoa" date="2017-12-13">adapted for new DataObject structure</revision>
       <revision author="alfoa" date="2018-04-26">Modified location of the test (ARMA folder)</revision>
       <revision author="talbpaul" date="2018-12-20">ROM IO to data object added</revision>
       <revision author="talbpaul" date="2018-09-13">changed Pmax and Pmin to P (same for Q)</revision>
+      <revision author="zhouj" date="2019-05-06">Modified location of the test to arma reseed </revision>
```
Comment (PaulTalbot-INL) [inline tests/framework/ROM/TimeSeries/ARMA/arma_reseed_precursor.xml]: And added sampling for the gold file as well as rommeta, right?

File: tests/framework/ROM/TimeSeries/ARMA/arma_reseed_test.xml [hunk 2]
```diff
@@ -6,36 +6,31 @@
     <created>2017-07-12</created>
     <classesTested>SupervisedLearning.ARMA</classesTested>
     <description>
-      Runs an unpickled ARMA ROM in internal parallel and assures the histories
-      generated are not identical (expected_fail = True).  Note that the gold file
-      is created by coping ARMAparallel/results_1.csv to gold/ARMAparallel/results_0.csv,
-      then we make sure they are not the same through expected_fail
+      Runs an unpickled ARMA ROM in internal parallel and assures the histories generated are
```
Comment (PaulTalbot-INL) [inline tests/framework/ROM/TimeSeries/ARMA/arma_reseed_test.xml]: Except we changed this to be NOT internal parallel, right? We can just remove "in internal parallel" here.

File: tests/framework/ROM/TimeSeries/ARMA/arma_reseed_test.xml [hunk 3]
```diff
@@ -6,36 +6,31 @@
...
+      identical (by setting the batch size to 1 and internalparelle to False in precursor and
+      reseed test). Note that the gold fileis created by run the arma_ressed_precurssor test,
```
Comment (PaulTalbot-INL) [inline tests/framework/ROM/TimeSeries/ARMA/arma_reseed_test.xml]: fileis -> file is

run -> running

File: tests/framework/ROM/TimeSeries/ARMA/tests [hunk 4]
```diff
@@ -25,15 +25,53 @@
...
+      unordered = true
+      output = 'ARMAreseed/rommeta.xml'
+      rel_err = 1e-5
+    [../]
+    [./signalexist]
+      type = Exists
+      output = 'ARMAreseed/signals_0.csv ARMAreseed/signals_1.csv ARMAreseed/signals_2.csv ARMAreseed/signals_3.csv ARMAreseed/signals.csv '
+    [../]
   [../]
   [./ARMAreseedTest]
```
Comment (PaulTalbot-INL) [inline tests/framework/ROM/TimeSeries/ARMA/tests]: shouldn't this test have ARMAreseedPrecursor as a prerequisite?

File: doc/user_manual/rom.tex [hunk 5]
```diff
@@ -2986,9 +2986,14 @@ \subsubsection{ARMA}
...
 \begin{equation*}
-x_t = y_t - \sum_m\left\{a_m\sin(2\pi f_mt)+b_m\cos(2\pi f_mt)\right\},
+\begin{aligned}
+x_t &= y_t - \sum_m\left\{a_m\sin(2\pi f_mt)+b_m\cos(2\pi f_mt)\right\}  \\
+&=y_t - \sum_m\ c_m\sin(2\pi f_mt+\phi_m)
+\end{aligned}
 \end{equation*}
-where $1/f_m$ is defined by the user parameter \xmlNode{Fourier}.
+
+where $1/f_m$ is defined by the user parameter \xmlNode{Fourier}. \nb $a_m$ and $b_m$ will be calculated then transform to
```
[the remark this answers is not in the document]
Comment (JiaZhou-PU) [reply doc/user_manual/rom.tex]: I think it's OK, just one line

**Comment by PaulTalbot-INL** [comment]
Is there an issue for this merge request? I think there is at least one that may be referenced if not closed by this.

**Comment by JiaZhou-PU** [comment]
Self-checked No.7

**Comment by PaulTalbot-INL** [comment]
An issue has been referenced; even though it's already closed, this further increases the testing for this feature and improves its coverage.

Checklists all pass, merging.

**Pull request merged**